api/src/auth/drivers/oauth2.ts

```diff
@@ -92,7 +92,7 @@ export class OAuth2AuthDriver extends LocalAuthDriver {
 		const user = await this.knex
 			.select('id')
 			.from('directus_users')
-			.whereRaw('LOWER(??) = ?', ['external_identifier', identifier.toLowerCase()])
+			.whereRaw('LOWER(??) = ?', ['external_identifier', identifier.toString().toLowerCase()])
```

Comment:
```suggestion
			.whereRaw('LOWER(??) = ?', ['external_identifier', String(identifier).toLowerCase()])
```

This should actually be done on line 135 since the signature for this function expects a string. Also, if you could apply this same fix to the openid driver it would be appreciated. 🙏

Reply:
Thanks for the suggestion, 
In the new commit toString() is called before fetchUserId
Also same fix applied for openId driver